numojo/core/array_creation_routines.mojo

```diff
@@ -408,7 +408,7 @@ fn empty[dtype: DType](*shape: Int) raises -> NDArray[dtype]:
     Returns:
         A NDArray of `dtype` with given `shape`.
     """
-    return NDArray[dtype](shape, fill=0)
+    return NDArray[dtype](shape=shape)
 
 
 fn zeros[dtype: DType](*shape: Int) raises -> NDArray[dtype]:
```

Comment:
This would also be randomly initialized from whatever was in the memory before like the `empty` one, fill should be set to 0.

Reply:
@MadAlex1997 But we are using `memset_zero` by default in all constructors and that should make all entries to be 0 even if we don't provide `fill=0`.

Reply:
@shivasankarka Then wouldn't empty just give the same result as zeros (an array memset to zero)? And wouldn't that represent unnecessary operations and therefore time (especially if the array is huge) if some other function was going to fill the array with values anyway?

If we are filling the array by an assignment in a loop instead of by setting it to zero and then adding a value to the array (which is vectorized by default), then we are not taking advantage of SIMD optimizations.

Reply:
@MadAlex1997 Yeah right, In case of `empty`, We need to remove the `memset_zero` since we gotta return unintialized array. 

I don't understand the second part, do you mean to say that we should use vectorized loop for filling values in the array?

Reply:
@shivasankarka If we initialize then memset zero and do a `+= fill_value` the `__iadd__` operation is SIMD vectorized by default. 


The other option which may be slightly faster would be to do the fill operation with SIMD vectorization in the initializers with fill values without memsetting to zero, as that should theoretically save the time for both the memset op and the add op.

Reply:
@MadAlex1997 Right, got it. I will make the change in this PR.